sdk/src/account_context.rs

```diff
@@ -0,0 +1,258 @@
+//! Successors of invoke_context::StackFrame, KeyedAccount and AccountInfo
+
+use crate::{
+    account::{AccountSharedData, ReadableAccount, WritableAccount},
+    instruction::InstructionError,
+    pubkey::Pubkey,
+};
+use std::cell::{RefCell, RefMut};
+
+/// Loaded transaction shared between runtime and programs.
+///
+/// This context is valid for the entire duration of a transaction being processed.
+pub struct TransactionContext {
+    account_keys: Vec<Pubkey>,
+    accounts: Vec<RefCell<AccountSharedData>>,
+    invoke_stack_max_height: usize,
+    invoke_stack: Vec<InstructionContext>,
+    return_data: (Pubkey, Vec<u8>),
+}
+
+impl TransactionContext {
+    pub fn new(
+        keys_and_accounts: Vec<(Pubkey, AccountSharedData)>,
+        invoke_stack_max_height: usize,
+    ) -> Self {
+        let (account_keys, accounts) = keys_and_accounts
+            .into_iter()
+            .map(|(key, account)| (key, RefCell::new(account)))
+            .unzip();
+        Self {
+            account_keys,
+            accounts,
+            invoke_stack_max_height,
+            invoke_stack: Vec::with_capacity(invoke_stack_max_height),
+            return_data: (Pubkey::default(), Vec::new()),
+        }
+    }
+
+    pub fn deconstruct(self) -> Vec<AccountSharedData> {
```

Comment:
When is this used?

Reply:
Will replace this part in the Bank:
https://github.com/solana-labs/solana/blob/ae024f666ade0ed30a8f2e0c6d946ec32a187dc0/runtime/src/bank.rs#L3591

The other half here is also going to be unnecessary as well:
https://github.com/solana-labs/solana/blob/ae024f666ade0ed30a8f2e0c6d946ec32a187dc0/runtime/src/bank.rs#L3532